plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/DeltaLakeQueryRunner.java

```diff
@@ -52,6 +54,73 @@
 
     private DeltaLakeQueryRunner() {}
 
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    public static Builder builder(Session defaultSession)
+    {
+        return new Builder(defaultSession);
+    }
+
+    public static class Builder
+            extends DistributedQueryRunner.Builder<Builder>
+    {
+        private String catalogName = DELTA_CATALOG;
+        private ImmutableMap.Builder<String, String> deltaProperties = ImmutableMap.builder();
+
+        protected Builder()
+        {
+            super(createSession());
+        }
+
+        protected Builder(Session defaultSession)
+        {
+            super(defaultSession);
+        }
+
+        public Builder setCatalogName(String catalogName)
+        {
+            this.catalogName = catalogName;
+            return self();
+        }
+
+        public Builder setDeltaProperties(Map<String, String> deltaProperties)
+        {
+            this.deltaProperties = ImmutableMap.<String, String>builder()
+                    .putAll(requireNonNull(deltaProperties, "deltaProperties is null"));
+            return self();
+        }
+
+        @Override
+        public DistributedQueryRunner build()
+                throws Exception
+        {
+            DistributedQueryRunner queryRunner = super.build();
+            try {
+                queryRunner.installPlugin(new TpchPlugin());
+                queryRunner.createCatalog("tpch", "tpch");
+
+                queryRunner.installPlugin(new TestingDeltaLakePlugin());
+                Map<String, String> deltaProperties = new HashMap<>(this.deltaProperties.buildOrThrow());
+                if (!deltaProperties.containsKey("hive.metastore.uri")) {
```

Comment:
Having business logic in setting up default values for properties is not a  good idea.
In case we'll want to handle Glue here, we'll further complicate the logic. 
I'd advocate that the client classes of `DeltaQueryRunner` should specify these properties themselves.

See for reference: 
- https://github.com/trinodb/trino/blob/74b324cf0089108492619a5176240c2ba2c8f36a/plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeSharedFileMetastoreWithTableRedirections.java#L49-L50
- https://github.com/trinodb/trino/blob/74b324cf0089108492619a5176240c2ba2c8f36a/plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeSharedGlueMetastoreWithTableRedirections.java#L75-L76

Reply:
I tend to agree with you. Seems like the `IcebergQueryRunner` also does this though. I would still consider putting it in the QueryRunner class itself as duplicating this logic everywhere seems a bit redundant. WDYT of adding a builder method `withFileMetastore`? Seems more explicit and hides the complexity for the clients of the QueryRunner classes.

Reply:
I would say, let's keep this out of this PR for now.

Reply:
I'm fine with `withFileMetastore` because it abstracts the logic of setting the file hive metastore.

> let's keep this out of this PR for now.
I would argue that this change is related to the current PR.

Reply:
Follow-up PR is acceptable to me. Going to merger as-is.